src/Text/Pandoc/Lua/Init.hs

```diff
@@ -78,23 +78,43 @@ initLuaState = do
         _                  -> T.pack $ show err
 
   setGlobalModules :: PandocLua ()
-  setGlobalModules = liftPandocLua $
-    forM_ [ ("lpeg", LPeg.luaopen_lpeg_ptr)
-          , ("re", LPeg.luaopen_re_ptr)
-          ] $
+  setGlobalModules = liftPandocLua $ do
+    let globalModules =
+          [ ("lpeg", LPeg.luaopen_lpeg_ptr)  -- must be loaded first
+          , ("re", LPeg.luaopen_re_ptr)      -- re depends on lpeg
+          ]
+    loadedInternal <- forM globalModules $
       \(pkgname, luaopen) -> do
         Lua.pushcfunction luaopen
-        Lua.pcall 0 1 Nothing >>= \case
-          OK -> pure ()          -- all good, loading succeeded
+        usedInternal <- Lua.pcall 0 1 Nothing >>= \case
+          OK -> do               -- all good, loading succeeded
+            -- register as loaded module so later modules can rely on this
+            Lua.getfield Lua.registryindex Lua.loaded
+            Lua.pushvalue (Lua.nth 2)
+            Lua.setfield (Lua.nth 2) pkgname
+            Lua.pop 1  -- pop _LOADED
+            return True
           _  -> do               -- built-in library failed, load system lib
             Lua.pop 1  -- ignore error message
             -- Try loading via the normal package loading mechanism.
             Lua.getglobal "require"
             Lua.pushName pkgname
             Lua.call 1 1  -- Throws an exception if loading failed again!
+            return False
 
         -- Module on top of stack. Register as global
         Lua.setglobal pkgname
+        return usedInternal
```

Comment:
How about
```
return (usedInternal, pkgname)
```
and then avoid the `zip` on line 112 and the `_` on 113?

Reply:
Or, perhaps even clearer, return `Just pkgname` instead of True and `Nothing` instead of False.

Reply:
Much nicer. :+1: